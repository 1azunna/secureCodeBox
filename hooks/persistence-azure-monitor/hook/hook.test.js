// SPDX-FileCopyrightText: 2022 iteratec GmbH
//
// SPDX-License-Identifier: Apache-2.0

process.env = Object.assign(process.env, {"MONITOR_WORKSPACE_ID": '123123', 
                                          "MONITOR_SHARED_KEY": "ffffffffffffffffffffffffffffff",
                                          "MONITOR_LOGTYPE_PREFIX": "SCB"});

const { handle } = require("./hook");

// Mock the fetch function
const fetch = jest.fn(x => new Promise((resolve, reject) => resolve({status: 200})));

beforeEach(() => {
  jest.clearAllMocks();
});

const scan = {
  metadata: {
    uid: "09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
    name: "demo-scan",
    labels: {
      company: "iteratec",
    },
  },
  spec: {
    scanType: "Nmap",
    parameters: ["-Pn", "localhost"],
  },
};

const testDate = new Date('2020-11-11');

test("should send findings to Azure Monitor", async () => {

  const findings = [
    {
      id: "4560b3e6-1219-4f5f-9b44-6579f5a32407",
      name: "Port 5601 is open",
      category: "Open Port",
    },
  ];

  const getFindings = async () => findings;

  await handle({getFindings,
                scan,
                workspaceId: "123123",
                sharedKey: "ffffffffffffffffffffffffffffff",
                logTypePrefix: "SCB",
                fetch: fetch,
                });

  expect(fetch).toBeCalledTimes(1);
});

test("should batch multiple findings in a single call", async () => {

  const findings = [
    {
      id: "4560b3e6-1219-4f5f-9b44-6579f5a32407",
      name: "Port 5601 is open",
      category: "Open Port",
    },
    {
      id: "4560b3e6-1219-4f5f-9b44-6579f5a32407",
      name: "Port 5601 is open",
      category: "Open Port",
    },
  ];

  const getFindings = async () => findings;

  await handle({getFindings,
                scan,
                workspaceId: "123123",
                sharedKey: "ffffffffffffffffffffffffffffff",
                logTypePrefix: "SCB",
                fetch: fetch,
                });

  expect(fetch).toBeCalledTimes(1);
});
