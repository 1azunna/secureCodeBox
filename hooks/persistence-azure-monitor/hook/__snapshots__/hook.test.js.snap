// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should batch multiple findings in a single call 1`] = `
Array [
  Array [
    "https://123123.ods.opinsights.azure.com/api/logs?api-version=2016-04-01",
    Object {
      "body": "[{\\"id\\":\\"4560b3e6-1219-4f5f-9b44-6579f5a32407\\",\\"name\\":\\"Port 5601 is open\\",\\"category\\":\\"Open Port\\",\\"scan_id\\":\\"09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc\\",\\"scan_name\\":\\"demo-scan\\",\\"scan_type\\":\\"Nmap\\",\\"scan_parameters\\":[\\"-Pn\\",\\"localhost\\"],\\"scan_labels\\":{\\"company\\":\\"iteratec\\"}},{\\"id\\":\\"4560b3e6-1219-4f5f-9b44-6579f5a32407\\",\\"name\\":\\"Port 5601 is open\\",\\"category\\":\\"Open Port\\",\\"scan_id\\":\\"09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc\\",\\"scan_name\\":\\"demo-scan\\",\\"scan_type\\":\\"Nmap\\",\\"scan_parameters\\":[\\"-Pn\\",\\"localhost\\"],\\"scan_labels\\":{\\"company\\":\\"iteratec\\"}}]",
      "headers": Object {
        "Authorization": "SharedKey 123123:BeWwef01Uml5v56eoq4+PFkZ/FxkwNbJ2wGwzD0A6Rg=",
        "Log-Type": "SCB_Nmap",
        "content-type": "application/json",
        "x-ms-date": 2020-11-11T00:00:00.000Z,
      },
      "method": "POST",
    },
  ],
]
`;

exports[`should pull settings from env variables 1`] = `
Array [
  Array [
    "https://123123.ods.opinsights.azure.com/api/logs?api-version=2016-04-01",
    Object {
      "body": "[{\\"id\\":\\"4560b3e6-1219-4f5f-9b44-6579f5a32407\\",\\"name\\":\\"Port 5601 is open\\",\\"category\\":\\"Open Port\\",\\"scan_id\\":\\"09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc\\",\\"scan_name\\":\\"demo-scan\\",\\"scan_type\\":\\"Nmap\\",\\"scan_parameters\\":[\\"-Pn\\",\\"localhost\\"],\\"scan_labels\\":{\\"company\\":\\"iteratec\\"}}]",
      "headers": Object {
        "Authorization": "SharedKey 123123:hYnj/mUzwD23iO1yAmgDoZcLu7/palVaaxiUgTCr1xQ=",
        "Log-Type": "SCB_Nmap",
        "content-type": "application/json",
        "x-ms-date": 2020-11-11T00:00:00.000Z,
      },
      "method": "POST",
    },
  ],
]
`;

exports[`should send findings to Azure Monitor 1`] = `
Array [
  Array [
    "https://123123.ods.opinsights.azure.com/api/logs?api-version=2016-04-01",
    Object {
      "body": "[{\\"id\\":\\"4560b3e6-1219-4f5f-9b44-6579f5a32407\\",\\"name\\":\\"Port 5601 is open\\",\\"category\\":\\"Open Port\\",\\"scan_id\\":\\"09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc\\",\\"scan_name\\":\\"demo-scan\\",\\"scan_type\\":\\"Nmap\\",\\"scan_parameters\\":[\\"-Pn\\",\\"localhost\\"],\\"scan_labels\\":{\\"company\\":\\"iteratec\\"}}]",
      "headers": Object {
        "Authorization": "SharedKey 123123:hYnj/mUzwD23iO1yAmgDoZcLu7/palVaaxiUgTCr1xQ=",
        "Log-Type": "SCB_Nmap",
        "content-type": "application/json",
        "x-ms-date": 2020-11-11T00:00:00.000Z,
      },
      "method": "POST",
    },
  ],
]
`;
